raiden_contracts/deploy/__main__.py

```diff
@@ -1,45 +1,106 @@
 """
 A simple Python script to deploy compiled contracts.
 """
+import json
 import logging
 from logging import getLogger
-from getpass import getpass
 
 import click
-import json
-
-from raiden_contracts.contract_manager import ContractManager, CONTRACTS_SOURCE_DIRS
-from web3 import Web3, HTTPProvider
-from eth_utils import (
-    is_address,
-    to_checksum_address,
-)
-
-from raiden_contracts.utils.utils import (
-    check_succesful_tx,
-)
-from raiden_contracts.constants import (
-    CONTRACT_ENDPOINT_REGISTRY,
-    CONTRACT_TOKEN_NETWORK_REGISTRY,
-    CONTRACT_SECRET_REGISTRY,
-    CONTRACT_CUSTOM_TOKEN,
-    DEPLOY_SETTLE_TIMEOUT_MIN,
-    DEPLOY_SETTLE_TIMEOUT_MAX,
-)
+from eth_utils import denoms, encode_hex, is_address, to_checksum_address
+from raiden_contracts.constants import (CONTRACT_CUSTOM_TOKEN,
+                                        CONTRACT_ENDPOINT_REGISTRY,
+                                        CONTRACT_SECRET_REGISTRY,
+                                        CONTRACT_TOKEN_NETWORK_REGISTRY,
+                                        DEPLOY_SETTLE_TIMEOUT_MAX,
+                                        DEPLOY_SETTLE_TIMEOUT_MIN)
+from raiden_contracts.contract_manager import (CONTRACTS_SOURCE_DIRS,
+                                               ContractManager)
+from raiden_contracts.utils.utils import check_succesful_tx
+
+from raiden_libs.private_contract import PrivateContract
+from raiden_libs.utils import get_private_key, private_key_to_address
+from web3 import HTTPProvider, Web3
 from web3.middleware import geth_poa_middleware
 
 log = getLogger(__name__)
 
 
-@click.command()
+def validate_address(ctx, param, value):
+    if not value:
+        return None
+    try:
+        is_address(value)
+        return to_checksum_address(value)
+    except ValueError:
+        raise click.BadParameter('must be a valid ethereum address')
+
+
+class ContractDeployer:
+    def __init__(
+        self,
+        web3,
```

Comment:
Since we're improving this code... would you be so kind and add type annotations here and in the methods below?